lib/good_job/active_job_extensions/concurrency.rb

```diff
@@ -19,6 +19,9 @@ module Concurrency
           key = job.good_job_concurrency_key
           next(block.call) if key.blank?
 
+          # Don't attempt to enforce concurrency limits with other queue adapters.
+          next(block.call) unless job.queue_adapter.is_a?(ActiveJob::QueueAdapters::GoodJobAdapter)
```

Comment:
I think queue_adapter lives on the class rather than the instance:`job.class.queue_adapter`.

Also, I think this should be the first guard clause inside of the `around_enqueue` block.

Reply:
Thanks—I moved the guards up to be the first check!

It looks like `queue_adapter` on instances delegates to the class, which was working for me on Rails 6.1, but maybe that’s a recent addition—I’ll see if `job.class.queue_adapter` fixes those test failures.